hibernate-core/src/test/java/org/hibernate/orm/test/idbag/IdBagTest.java

```diff
@@ -76,33 +73,33 @@ public void testUpdateIdBag() throws HibernateException {
 	}
 
 	@Test
-	public void testJoin() throws HibernateException, SQLException {
-		Session s = openSession();
-		Transaction t = s.beginTransaction();
-		User gavin = new User( "gavin" );
-		Group admins = new Group( "admins" );
-		Group plebs = new Group( "plebs" );
-		gavin.getGroups().add( plebs );
-		gavin.getGroups().add( admins );
-		s.persist( gavin );
-		s.persist( plebs );
-		s.persist( admins );
-
-		List l = s.createQuery( "from User u join u.groups g" ).list();
-		assertEquals( l.size(), 2 );
-		s.clear();
-
-		gavin = (User) s.createQuery( "from User u join fetch u.groups" ).uniqueResult();
-		assertTrue( Hibernate.isInitialized( gavin.getGroups() ) );
-		assertEquals( gavin.getGroups().size(), 2 );
-		assertEquals( ( (Group) gavin.getGroups().get( 0 ) ).getName(), "admins" );
-
-		s.delete( gavin.getGroups().get( 0 ) );
-		s.delete( gavin.getGroups().get( 1 ) );
-		s.delete( gavin );
-
-		t.commit();
-		s.close();
+	public void testJoin(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					User gavin = new User( "gavin" );
+					Group admins = new Group( "admins" );
+					Group plebs = new Group( "plebs" );
+					gavin.getGroups().add( plebs );
+					gavin.getGroups().add( admins );
+					session.persist( gavin );
+					session.persist( plebs );
+					session.persist( admins );
+
+					List l = session.createQuery( "from User u join u.groups g" ).list();
+					assertEquals( 1, l.size() );
+
+					session.clear();
+
+					gavin = (User) session.createQuery( "from User u join fetch u.groups" ).uniqueResult();
+					assertTrue( Hibernate.isInitialized( gavin.getGroups() ) );
+					assertEquals( gavin.getGroups().size(), 2 );
+					assertEquals( ( (Group) gavin.getGroups().get( 0 ) ).getName(), "admins" );
```

Comment:
technically, you messed up the parameter order, though the difference can only be noticed when the testing case is broken.

Reply:
I haven't noticed that the original was using the wrong order, I'm going to fix it.
Thanks